clients/src/main/java/org/apache/kafka/clients/consumer/internals/Coordinator.java

```diff
@@ -460,6 +460,12 @@ private void maybeAutoCommitOffsetsSync() {
                     offsetAndMetadata.offset(), offsetAndMetadata.metadata()));
         }
 
+        // reset generation/consumerId to default values if using manual assignment
+        if (!subscriptions.partitionsAutoAssigned()) {
+            this.consumerId = OffsetCommitRequest.DEFAULT_CONSUMER_ID;
```

Comment:
I'm not sure if it's right to replace the check with just the reset in KafkaConsumer.unsubscribe() since users are not guaranteed to call KafkaConsumer.unsubscribe().

Isn't @guozhangwang's point when users cross over from different subscription modes still an issue?